modules/gapi/test/util/variant_tests.cpp

```diff
@@ -486,4 +500,209 @@ TEST(Variant, EXT_IndexOf)
     static_assert(6u == V::index_of<MyClass>(), "Index is incorrect");
 }
 
+namespace test_validation
+{
+struct MyType
+{
+    friend std::ostream& operator<<(std::ostream& out, const MyType& src)
+    {
+        return out << "MyType"; (void) src;
+    }
+};
+class MyClass
+{
+    friend std::ostream& operator<<(std::ostream& out, const MyClass& src)
+    {
+        return out << "MyClass"; (void) src;
+    }
+};
+
+struct MyParamDynamicVisitor : cv::util::static_indexed_visitor<bool, MyParamDynamicVisitor>
+{
+    MyParamDynamicVisitor(std::ostream &output) : out(output) {}
+
+    template<class Type>
+    bool visit(std::size_t index, Type val, int check)
+    {
+        bool result = false;
+        out << index << ":" << val <<",";
+        if(std::is_same<Type, int>::value)
+        {
+            result = !memcmp(&val, &check, sizeof(int));
+        }
+        return result;
+    }
+
+    std::ostream &out;
+};
+
+struct MyNoParamDynamicVisitor : cv::util::static_indexed_visitor<bool, MyNoParamDynamicVisitor>
+{
+    MyNoParamDynamicVisitor(std::ostream &output) : out(output) {}
+
+    template<class Type>
+    bool visit(std::size_t index, Type val)
+    {
+        out << index << ":" << val <<",";
+        return true;
+    }
+    std::ostream &out;
+};
+
+
+struct MyNoParamStaticVisitor : cv::util::static_visitor<void, MyNoParamStaticVisitor>
```

Comment:
Also, what's the difference between Static and Dynamic visitors here?

Reply:
This class implements  `visit(Type val)`, but Dynamic implements `visit(std::size_t index, Type val`

I agree  that class name is confused, i will rename it

Reply:
What's the difference between those and should user care about that?

Reply:
the second version `visit(std::size_t index, Type val)`  for `indexed_visitor` returns not only value from variant but also its position index and provide additional flexibility in processing the value (  for example logging, printing, or more compilex use case in serialization maybe)